### PR TITLE
Allow to specify ISO checksum via URL

### PR DESCRIPTION
**Pull request by rickard-von-essen**
See #2744

### CODE REVIEW

**Comment by rickard-von-essen** [comment]
@vtolstov Can you quickly review this? And if you have time test it for some of your use cases.

File: common/iso_config.go [hunk 1]
```diff
@@ -23,36 +30,81 @@ func (c *ISOConfig) Prepare(ctx *interpolate.Context) ([]string, []error) {
...
+				// If iso_checksum has no value use iso_checksum_url instead.
+				if c.ISOChecksum == "" {
+					u, err := url.Parse(c.ISOChecksumURL)
+					if err != nil {
+						errs = append(errs,
+							fmt.Errorf("Error parsing checksum: %s", err))
+						return warnings, errs
+					}
+					switch u.Scheme {
+					case "http", "https", "ftp", "ftps":
```
Comment (vtolstov) [inline common/iso_config.go]: May be drop ftp, ftps ? I don't try this scheme via http.Get and it may not work..

**Comment by vtolstov** [comment]
LGTM, with minor comment about `case "http", "https", "ftp", "ftps":`

**Comment by vtolstov** [comment]
Thanks!

**Comment by vtolstov** [comment]
I'm test this pr, nothing breaks for me, so i think it fine.

**Comment by rickard-von-essen** [comment]
@vtolstov I tested ftp but it didn't work so I removed ftp/ftps. I don't think they are very important.

**Comment by vtolstov** [comment]
@rickard-von-essen thanks! so as i think this is ready to merge =)

**Pull request merged**